src/hooks/useCheckboxHandler.ts

```diff
@@ -0,0 +1,50 @@
+import { useEffect, useMemo, useRef, useState } from "react";
+import { CartItemType } from "../types/response";
+
+const useCheckboxHandler = (cartItems: CartItemType[]) => {
+  const allCartIds = useMemo(
+    () => [...cartItems].map((item) => item.id),
```

Comment:
map은 새로운 객체를 반환해서 그대로 사용해도 괜찮겠네요!
```suggestion
    () => cartItems.map((item) => item.id),
```

Reply:
selectedCartIds의 초기값으로 전체 id를 설정하는 로직을 개선하면서, cartItems로 allCartIds 계산을 바로 수행하도록 변경했습니다!

[refactor: 초기 화면에서 전체 선택이 체크되는 로직을 개선 (73ebf0d)](https://github.com/woowacourse/react-shopping-cart/pull/354/commits/73ebf0d969fd813123f9c6c5b20493a276b70713)